iree-prof-tools/graph.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_PROF_GRAPH_H_
+#define IREE_PROF_GRAPH_H_
+
+#include <string>
+
+#include "third_party/llvm-project/llvm/include/llvm/Support/JSON.h"
+
+namespace iree_prof::graph {
+
+struct Attribute {
+  Attribute(std::string key, std::string value)
+      : key(std::move(key)), value(std::move(value)) {}
+  std::string key;
+  std::string value;
+
+  llvm::json::Object Json();
+};
+
+struct Metadata {
+  std::string id;
+  std::vector<Attribute> attrs;
+
+  llvm::json::Object Json();
+};
+
+struct GraphEdge {
+  std::string source_node_id;
```

Comment:
Isn't it supposed to be an edge ID? It is interesting to keep the source ID.
You may add comment to explain those three IDs.

Reply:
It's correct according to google internal source. Copied comments of all structs from the internal source.